eos/form-factors/parametric-bfw2010-impl.hh

```diff
@@ -0,0 +1,808 @@
+    }
+    template<typename Process_>
+    inline double
+    BFW2010FormFactors<Process_, PToV>::_phi_a_0(const double & q2) const
+    {
+        return _phi(q2, _tp_a, Process_::chi_0m_a, 0, 0, 0, 4, 2, 1, 3);
+    }
+    template<typename Process_>
+    inline double
+    BFW2010FormFactors<Process_, PToV>::_phi_a_1(const double & q2) const
+    {
+        return _phi(q2, _tp_a, Process_::chi_1p_a, 1, 0, 0, 6, 1, 2, 1);
+    }
+    template<typename Process_>
+    inline double
+    BFW2010FormFactors<Process_, PToV>::_phi_a_12(const double & q2) const
+    {
+        return _phi(q2, _tp_a, Process_::chi_1p_a, 0, 0, 2, 3, 2, 2, 1);
+    }
+    template<typename Process_>
+    inline double
+    BFW2010FormFactors<Process_, PToV>::_phi_t_1(const double & q2) const
+    {
+        return _phi(q2, _tp_v, Process_::chi_1m_t, 0, 0, 0, 6, 1, 3, 3);
+    }
+    template<typename Process_>
+    inline double
+    BFW2010FormFactors<Process_, PToV>::_phi_t_2(const double & q2) const
+    {
+        return _phi(q2, _tp_a, Process_::chi_1p_t5, 1, 1, 0, 6, 1, 3, 1);
+    }
+    template<typename Process_>
+    inline double
+    BFW2010FormFactors<Process_, PToV>::_phi_t_23(const double & q2) const
+    {
+        return _phi(q2, _tp_a, Process_::chi_1p_t5, -1, 0, 2, 12, 0, 3, 1);
+    }
+
+
+    template <typename Process_>
+    double
+    BFW2010FormFactors<Process_, PToV>::_a_A12_0() const
+    {
+        const double x_A12 = this->_traits.calc_z(0.0, _tp_a, _traits.m_R2_1p) * this->_phi_a_12(0.0) * (_mB2 - _mV2) / 8.0 / _mB / _mV;
+        const double x_A0  = this->_traits.calc_z(0.0, _tp_a, _traits.m_R2_0m) * this->_phi_a_0(0.0);
+        std::array<double, 5> a;
+        a[0] = x_A12 * this->_a_A0[0];
+        for (unsigned i = 1 ; i < a.size() ; ++i)
+        {
+            a[i] = x_A12 * this->_a_A0[i] - x_A0 * this->_a_A12[i - 1];
+        }
+        const auto polynomials = Process_::orthonormal_polynomials_a(_traits.calc_z(0.0, _tp_a, _t0));
+        return std::inner_product(a.begin(), a.end(), polynomials.begin(), 0.0) / (polynomials[0] * x_A0);
+    }
+
+    template <typename Process_>
+    double
+    BFW2010FormFactors<Process_, PToV>::_a_T2_0() const
+    {
+        const double x_T2 = this->_traits.calc_z(0.0, _tp_a, _traits.m_R2_1p) * this->_phi_t_2(0.0);
+        const double x_T1 = this->_traits.calc_z(0.0, _tp_v, _traits.m_R2_1m) * this->_phi_t_1(0.0);
+        const auto polynomials_T2 = Process_::orthonormal_polynomials_a(_traits.calc_z(0.0, _tp_a, _t0));
+        const auto polynomials_T1 = Process_::orthonormal_polynomials_v(_traits.calc_z(0.0, _tp_v, _t0));
+
+        double a_T2_0 = x_T2 * this->_a_T1[0] * polynomials_T1[0];
+        for (unsigned i = 1 ; i < _a_T1.size() ; ++i)
+        {
+            a_T2_0 += x_T2 * this->_a_T1[i] * polynomials_T1[i] - x_T1 * this->_a_T2[i - 1] * polynomials_T2[i];
+        }
+
+        return a_T2_0 / (polynomials_T2[0] * x_T1);
+    }
+
```

Comment:
Superfluous empty line.

Reply:
Well, it was separating the end-point relations and the form-factor implementations. But I removed it